res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+        'fx2on': 0x48,
+        'shiftFx2on': 0x64,
+        'fx3on': 0x49,
+        'shiftFx3on': 0x65,
+        'fxTab': 0x4A,
+        'shiftFxTab': 0x66,
+        'fx1assignDeck1': 0x4C,
+        'shiftFx1assignDeck1': 0x70,
+        'fx1assignDeck2': 0x4D,
+        'shiftFx1assignDeck2': 0x71,
+        'fx1assignDeck3': 0x4E,
+        'shiftFx1assignDeck3': 0x72,
+        'fx1assignDeck4': 0x4F,
+        'shiftFx1assignDeck4': 0x73,
+        'fx2assignDeck1': 0x50,
+        'shiftFx2assignDeck1': 0x54,
+        'fx2assignDeck2': 0x51,
+        'shiftFx2assignDeck2': 0x55,
+        'fx2assignDeck3': 0x52,
+        'shiftFx2assignDeck3': 0x56,
+        'fx2assignDeck4': 0x53,
+        'shiftFx2assignDeck4': 0x57,
+        'masterCue': 0x63,
+        'shiftMasterCue': 0x62,
+        'loadDeck1': 0x46,
+        'shiftLoadDeck1': 0x58,
+        'loadDeck2': 0x47,
+        'shiftLoadDeck2': 0x59,
+        'loadDeck3': 0x48,
+        'shiftLoadDeck3': 0x60,
+        'loadDeck4': 0x49,
+        'shiftLoadDeck4': 0x61,
+        'hotCueMode': 0x1B,
+        'shiftHotCueMode': 0x69,
+        'rollMode': 0x1E,
+        'shiftRollMode': 0x6B,
+        'slicerMode': 0x20,
+        'shiftSlicerMode': 0x6D,
+        'samplerMode': 0x22,
+        'shiftSamplerMode': 0x6F,
+        'longPressSamplerMode': 0x41,
+        'parameterLeftHotCueMode': 0x24,
+        'shiftParameterLeftHotCueMode': 0x01,
+        'parameterLeftRollMode': 0x25,
+        'shiftParameterLeftRollMode': 0x02,
+        'parameterLeftSlicerMode': 0x26,
+        'shiftParameterLeftSlicerMode': 0x03,
+        'parameterLeftSamplerMode': 0x27,
+        'shiftParameterLeftSamplerMode': 0x04,
+        'parameterLeftGroup1Mode': 0x28,
+        'shiftParameterLeftGroup1Mode': 0x05,
+        'parameterLeftGroup2Mode': 0x29,
+        'shiftParameterLeftGroup2Mode': 0x06,
+        'parameterLeftGroup3Mode': 0x2A,
+        'shiftParameterLeftGroup3Mode': 0x07,
+        'parameterLeftGroup4Mode': 0x2B,
+        'shiftParameterLeftGroup4Mode': 0x08,
+        'parameterRightHotCueMode': 0x2C,
+        'shiftParameterRightHotCueMode': 0x09,
+        'parameterRightRollMode': 0x2D,
+        'shiftParameterRightRollMode': 0x7A,
+        'parameterRightSlicerMode': 0x2E,
+        'shiftParameterRightSlicerMode': 0x7B,
+        'parameterRightSamplerMode': 0x2F,
+        'shiftParameterRightSamplerMode': 0x7C,
+        'parameterRightGroup1Mode': 0x30,
+        'shiftParameterRightGroup1Mode': 0x7D,
+        'parameterRightGroup2Mode': 0x31,
+        'shiftParameterRightGroup2Mode': 0x7E,
+        'parameterRightGroup3Mode': 0x32,
+        'shiftParameterRightGroup3Mode': 0x7F,
+        'parameterRightGroup4Mode': 0x33,
+        'shiftParameterRightGroup4Mode': 0x00
+    };
+
+    PioneerDDJSX.illuminationControl = {
+        'loadedDeck': 0x00, //deck 1-4: 0x00 - 0x03
+        'playPauseDeck': 0x0C, //deck 1-4: 0x0C - 0x0F
+        'cueDeck': 0x10, //deck 1-4: 0x10 - 0x13
+        'djAppConnect': 0x09
```

Comment:
What does sending this MIDI message do? It does not look like you send it anywhere in the script. I am guessing that it asks the controller to send the positions of all the knobs and faders so their initial positions can be set in the software on startup. If so, you'll have to delay enabling soft takeover until after the first MIDI messages are received (the Components library does that automatically).